### PR TITLE
Fixes an issue where the unequipped items with options do not re-calculate status.

### PR DESCRIPTION
**Pull request by Smokexyz**
### Pull Request Prelude

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
Fixes an issue where the unequipped items with options do not re-calculate status.

**Affected Branches:** Master

**Issues addressed:** http://herc.ws/board/topic/14581-introducing-the-item-options-system/?p=81615

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/1673)

File: src/map/pc.c [hunk 1]
```diff
@@ -9986,51 +9986,51 @@ void pc_unequipitem_pos(struct map_session_data *sd, int n, int pos)
...
+	if (n < 0 || n >= MAX_INVENTORY) {
+		clif->unequipitemack(sd, 0, 0, UIA_FAIL);
 		return 0;
 	}
 
 	// if player is berserk then cannot unequip
-	if (!(flag&PCUNEQUIPITEM_FORCE) && sd->sc.count && (sd->sc.data[SC_BERSERK] || sd->sc.data[SC_NO_SWITCH_EQUIP]) )
-	{
-		clif->unequipitemack(sd,n,0,UIA_FAIL);
+	if (!(flag&PCUNEQUIPITEM_FORCE) && sd->sc.count && (sd->sc.data[SC_BERSERK] || sd->sc.data[SC_NO_SWITCH_EQUIP])) {
```
Comment (4144) [inline src/map/pc.c]: if you changing style on this line, probably also add missing spaces after/before &

File: src/map/pc.c [hunk 2]
```diff
@@ -9986,51 +9986,51 @@ void pc_unequipitem_pos(struct map_session_data *sd, int n, int pos)
...
+		clif->unequipitemack(sd, n, 0, UIA_FAIL);
 		return 0;
 	}
 
-	if( !(flag&PCUNEQUIPITEM_FORCE) && sd->sc.count && sd->sc.data[SC_KYOUGAKU] )
-	{
-		clif->unequipitemack(sd,n,0,UIA_FAIL);
+	if (!(flag&PCUNEQUIPITEM_FORCE) && sd->sc.count && sd->sc.data[SC_KYOUGAKU]) {
```
Comment (4144) [inline src/map/pc.c]: also add spaces here

File: src/map/pc.c [hunk 3]
```diff
@@ -9986,51 +9986,51 @@ void pc_unequipitem_pos(struct map_session_data *sd, int n, int pos)
...
 
-	if(!sd->status.inventory[n].equip){ //Nothing to unequip
-		clif->unequipitemack(sd,n,0,UIA_FAIL);
+	if (!sd->status.inventory[n].equip) { //Nothing to unequip
+		clif->unequipitemack(sd, n, 0, UIA_FAIL);
 		return 0;
 	}
-	for(i=0;i<EQI_MAX;i++) {
+
+	for (i=0;i<EQI_MAX;i++) {
```
Comment (4144) [inline src/map/pc.c]: add spaces here after/before = and <. and after ;

File: src/map/pc.c [hunk 4]
```diff
@@ -10043,70 +10043,80 @@ int pc_unequipitem(struct map_session_data *sd,int n,int flag)
...
+				continue;
+			if ((ito = itemdb->option_exists(sd->status.inventory[n].option[i].index)) == NULL)
+				continue;
+
+			status_calc = true;
+		}
 	}
 
-	if(flag&PCUNEQUIPITEM_RECALC || status_cacl) {
+	if (flag&PCUNEQUIPITEM_RECALC || status_calc) {
```
Comment (4144) [inline src/map/pc.c]: add spaces here before/after &

File: src/map/pc.c [hunk 5]
```diff
@@ -10043,70 +10043,80 @@ int pc_unequipitem(struct map_session_data *sd,int n,int flag)
...
 					script->run_item_unequip_script(sd, sd->inventory_data[n], npc->fake_nd->bl.id);
 			}
 			else
 				script->run_item_unequip_script(sd, sd->inventory_data[n], npc->fake_nd->bl.id);
 		}
-		if(itemdb_isspecial(sd->status.inventory[n].card[0]))
-			; //No cards
-		else {
-			for( i = 0; i < sd->inventory_data[n]->slot; i++ ) {
+		if (itemdb_isspecial(sd->status.inventory[n].card[0]) == false){
```
Comment (4144) [inline src/map/pc.c]: add space before {

File: src/map/pc.c [hunk 6]
```diff
@@ -9986,51 +9986,51 @@ void pc_unequipitem_pos(struct map_session_data *sd, int n, int pos)
...
-	if(!sd->status.inventory[n].equip){ //Nothing to unequip
-		clif->unequipitemack(sd,n,0,UIA_FAIL);
+	if (sd->status.inventory[n].equip == 0) { //Nothing to unequip
+		clif->unequipitemack(sd, n, 0, UIA_FAIL);
 		return 0;
 	}
-	for(i=0;i<EQI_MAX;i++) {
+
+	for (i = 0; i < EQI_MAX; i++) {
 		if(sd->status.inventory[n].equip & pc->equip_pos[i])
```
Comment (cjedzkie) [inline src/map/pc.c]: missing space between if and (

File: src/map/pc.c [hunk 7]
```diff
@@ -10039,23 +10039,23 @@ int pc_unequipitem(struct map_session_data *sd,int n,int flag)
 	if( sd->state.autobonus&pos )
```
Comment (cjedzkie) [inline src/map/pc.c]: if (sd->state.autobonus&pos)

File: src/map/pc.c [hunk 8]
```diff
@@ -10075,49 +10075,48 @@ int pc_unequipitem(struct map_session_data *sd,int n,int flag)
...
 		status_change_end(&sd->bl, SC_CRUCIS, INVALID_TIMER);
 
 	//OnUnEquip script [Skotlex]
-	if (sd->inventory_data[n]) {
-		if (sd->inventory_data[n]->unequip_script) {
-			if ( battle_config.unequip_restricted_equipment & 1 ) {
+	if (sd->inventory_data[n] != NULL) {
+		if (sd->inventory_data[n]->unequip_script != NULL) {
+			if (battle_config.unequip_restricted_equipment & 1) {
 				ARR_FIND(0, map->list[sd->bl.m].zone->disabled_items_count, i,  map->list[sd->bl.m].zone->disabled_items[i] == sd->status.inventory[n].nameid);
```
Comment (cjedzkie) [inline src/map/pc.c]: extra space after `map->list[sd->bl.m]`

File: src/map/pc.c [hunk 9]
```diff
@@ -10075,49 +10075,48 @@ int pc_unequipitem(struct map_session_data *sd,int n,int flag)
...
 					continue;
 
-				if ( ( data = itemdb->exists(sd->status.inventory[n].card[i]) ) != NULL ) {
-					if ( data->unequip_script ) {
-						if ( battle_config.unequip_restricted_equipment & 2 ) {
+				if ((data = itemdb->exists(sd->status.inventory[n].card[i])) != NULL) {
+					if (data->unequip_script) {
+						if (battle_config.unequip_restricted_equipment & 2) {
 							int j;
 							ARR_FIND(0, map->list[sd->bl.m].zone->disabled_items_count, j,  map->list[sd->bl.m].zone->disabled_items[j] == sd->status.inventory[n].card[i]);
```
Comment (cjedzkie) [inline src/map/pc.c]: extra space before `map->list[sd->bl.m]

File: src/map/pc.c (hunk 7 above)
Comment (Smokexyz) [inline src/map/pc.c]: This line can't be edited, to avoid conflicts.

File: src/map/pc.c [hunk 10]
```diff
@@ -10039,74 +10039,84 @@ int pc_unequipitem(struct map_session_data *sd,int n,int flag)
...
+						if (pc->removecombo(sd, data))
+							status_calc = true;
 					}
 				}
 			}
 		}
+		/* Item Options checking */
+		for (i = 0; i < MAX_ITEM_OPTIONS; i++) {
+			struct item_option *ito = NULL;
+			uint16 item_option = sd->status.inventory[n].option[i].index;
```
Comment (MishimaHaruna) [inline src/map/pc.c]: index is signed

Comment (Smokexyz) [inline src/map/pc.c]: Fixed!

File: src/map/clif.c [hunk 11]
```diff
@@ -6417,7 +6417,7 @@ void clif_openvending(struct map_session_data* sd, int id, struct s_vending* ven
 		WFIFOB(fd,21+i*item_length) = sd->status.cart[index].refine;
 		clif->addcards(WFIFOP(fd,22+i*item_length), &sd->status.cart[index]);
 #if PACKETVER >= 20150226
-		clif->add_item_options(WFIFOP(fd,30+22+i*item_length), &sd->status.cart[index]);
+		clif->add_item_options(WFIFOP(fd,30+i*item_length), &sd->status.cart[index]);
```
Comment (4144) [inline src/map/clif.c]: here style/
space after coma. before and after plus, etc.

Comment (Jedzkie) [inline src/map/clif.c]: `clif->add_item_options(WFIFOP(fd, 30 + i * item_length), &sd->status.cart[index]);`

**Comment by changelets** [comment]
I'm wondering when this will be merged because it contains a fix in this issue #1685

**Comment by MishimaHaruna** [comment]
Merged, thanks

**Pull request merged**